internal/app/machined/internal/phase/rootfs/etc/etc.go

```diff
@@ -85,19 +87,31 @@ func Hosts(hostname string) (err error) {
 // ResolvConf copies the resolv.conf generated in the early boot to the new
 // root.
 func ResolvConf() (err error) {
-	target := "/run/system/etc/resolv.conf"
+	option := kernel.ProcCmdline().Get("ip").First()
+	switch option {
+	case nil:
+		target := "/run/system/etc/resolv.conf"
 
-	var f *os.File
+		var f *os.File
 
-	if f, err = os.OpenFile(target, os.O_WRONLY|os.O_CREATE, 0644); err != nil {
-		return err
-	}
+		if f, err = os.OpenFile(target, os.O_WRONLY|os.O_CREATE, 0644); err != nil {
+			return err
+		}
+
+		// nolint: errcheck
+		defer f.Close()
 
-	// nolint: errcheck
-	defer f.Close()
+		if err = unix.Mount("/run/system/etc/resolv.conf", "/etc/resolv.conf", "", unix.MS_BIND, ""); err != nil {
+			return fmt.Errorf("failed to create bind mount for /etc/resolv.conf: %w", err)
+		}
+	default:
+		if _, err = os.Stat("/proc/net/pnp"); os.IsNotExist(err) {
+			return errors.New("failed to setup /etc/resolv.conf, /proc/net/pnp does not exist")
```

Comment:
this looks wrong as it ignores errors which are not `os.IsNotExist()`

Reply:
I would probably simply return any `err != nil` via `fmt.Errorf()` context

Reply:
Good point. It gets a little ugly because`!os.IsNotExist()` doesn't guarantee that it does exist because the error _could_ have been something else. I can try to rework it.

Reply:
I think it looks better now, but I would rather still keep original `err`, as it might be anything, with `fmt.Errorf`

Reply:
Whoops sorry, was rushing this as I was running out the door.